app/views/shared/_bar_chart.html.erb

```diff
@@ -0,0 +1,27 @@
+<%= content_tag :canvas, nil, id: "chart-for-#{problem.id}", width: width, height: height, data: { labels: bar_chart_x_axis_labels(show_labels).to_json, notice_count: data, show_labels: show_labels, scale_steps: bar_chart_scale_steps(data.max) } %>
+
+<%= javascript_tag do %>
+  var canvas = $("#chart-for-<%= problem.id %>")
```

Comment:
Absolutely. First, you'd need to somehow identify this canvas tag as a bar chart. That could be as simple as adding a class as in: ```<canvas class="bar-chart" data-problem-id="#{problem.id}"...>```

I'll just rough out a snippet that we could put in application.js or some similar place. This probably has errors/bugs because I'm just freehanding it, but this would make the js entirely static. The only dynamic parts are the data attributes of the canvas tag:
```javascript
$('canvas.bar-chart').each(function(){
  var canvas = $(this);
  var problemId = canvas.data('problemId');
  var ctx = canvas.get(0).getContext("2d");

  chart = new Chart(ctx);
  chart.Bar(data, {
    scaleShowLabels: canvas.data('show-labels'),
    scaleOverride: true,
    scaleSteps: canvas.data('scale-steps'),
    scaleStepWidth: 1,
    scaleStartValue: 0,
    animation: false,
    scaleShowGridLines: false,
    barValueSpacing: 1
  });
});
```